### ISSUE TITLE
update categorical annotation after subsetting

### ISSUE THREAD

**Issue by bobermayer** [scverse/scanpy#69]
following the [mouse atlas example](https://github.com/dpcook/fun_analysis/blob/master/tabula_muris/mouse_atlas_scanpy.ipynb), I found that when selecting a subset of cells

```python
mammary=adata[adata.obs['Tissue']=='Mammary']
```

categorical annotations (e.g., `CellType`) keep all their original levels (categories), leading to a lot of legend entries without corresponding dots when doing a tSNE plot afterwards. 

adding
```python
mammary.obs['CellType']=mammary.obs['CellType'].values.remove_unused_categories()
```
fixes the problem, but maybe there's a more elegant solution?

**Comment by falexwolf** [comment]
Hi Benedikt!

Yes, we could make this automatic. Let me briefly think whether this will have any unwanted side effects. There are some advantages of letting the user fully control the annotation dataframes.

PS: you could also call `mammary.obs['CellType'].cat.remove_unused_categories(inplace=True)`.
PPS: The mouse atlas example has been written by a first-time-python user, according to what he stated. It can be written a lot more elegantly. We will soon update it...

**Comment by bobermayer** [comment]
Hi Alex,

cool thanks. btw great package, so happy I can do this in python!

**Comment by falexwolf** [comment]
Thank you for the kind words... Scanpy will still improve a lot in the upcoming weeks and months...

**Comment by falexwolf** [comment]
This has been implemented in anndata 0.5 and scanpy 0.4.3. See release notes (https://scanpy.readthedocs.io). See https://github.com/theislab/anndata/commit/8cabf9c86a38d6db88c664e2ea28e3fb29bdf99e and a few fixes after that.
